extensions/aws/src/main/java/io/opentelemetry/extension/aws/AwsXrayPropagator.java

```diff
@@ -107,19 +109,30 @@ public static AwsXrayPropagator getInstance() {
     char samplingFlag = spanContext.isSampled() ? IS_SAMPLED : NOT_SAMPLED;
     // TODO: Add OT trace state to the X-Ray trace header
 
-    String traceHeader =
-        TRACE_ID_KEY
-            + KV_DELIMITER
-            + xrayTraceId
-            + TRACE_HEADER_DELIMITER
-            + PARENT_ID_KEY
-            + KV_DELIMITER
-            + parentId
-            + TRACE_HEADER_DELIMITER
-            + SAMPLED_FLAG_KEY
-            + KV_DELIMITER
-            + samplingFlag;
-    setter.set(carrier, TRACE_HEADER_KEY, traceHeader);
+    StringBuilder traceHeader = new StringBuilder();
+    traceHeader
+        .append(TRACE_ID_KEY)
+        .append(KV_DELIMITER)
+        .append(xrayTraceId)
+        .append(TRACE_HEADER_DELIMITER)
+        .append(PARENT_ID_KEY)
+        .append(KV_DELIMITER)
+        .append(parentId)
+        .append(TRACE_HEADER_DELIMITER)
+        .append(SAMPLED_FLAG_KEY)
+        .append(KV_DELIMITER)
+        .append(samplingFlag);
+
+    Baggage baggage = Baggage.fromContext(context);
+    baggage.forEach(
+        (key, value) ->
```

Comment:
any worries here about blowing out the header if there's a lot shoved into the baggage?

Reply:
I think similar to how tracestate is handled by tracestatebuilder, for better or worse, baggage would similar be constrained at that logic layer instead of propagation.